src/anyio/_core/_tempfile.py

```diff
@@ -0,0 +1,272 @@
+        self.mode = mode
+        self.buffering = buffering
+        self.encoding = encoding
+        self.newline = newline
+        self.suffix: AnyStr | None = suffix
+        self.prefix: AnyStr | None = prefix
+        self.dir: AnyStr | None = dir
+        self.delete = delete
+        self.errors = errors
+        self.delete_on_close = delete_on_close
+
+        self._async_file: AsyncFile | None = None
+
+    async def __aenter__(self) -> AsyncFile[AnyStr]:
+        params: dict[str, Any] = {
+            "mode": self.mode,
+            "buffering": self.buffering,
+            "encoding": self.encoding,
+            "newline": self.newline,
+            "suffix": self.suffix,
+            "prefix": self.prefix,
+            "dir": self.dir,
+            "delete": self.delete,
+            "errors": self.errors,
+        }
+        if sys.version_info >= (3, 12):
+            params["delete_on_close"] = self.delete_on_close
+
+        fp = await to_thread.run_sync(lambda: tempfile.NamedTemporaryFile(**params))
+
+        self._async_file = AsyncFile(fp)
+        return self._async_file
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_value: BaseException | None,
+        traceback: TracebackType | None,
+    ) -> None:
+        if self._async_file is not None:
+            await self._async_file.aclose()
+            self._async_file = None
+
+
+class SpooledTemporaryFile(Generic[AnyStr]):
+    def __init__(
+        self,
+        max_size: int = 0,
+        mode: str = "w+b",
+        buffering: int = -1,
+        encoding: str | None = None,
+        newline: str | None = None,
+        suffix: AnyStr | None = None,
+        prefix: AnyStr | None = None,
+        dir: AnyStr | None = None,
+        *,
+        errors: str | None = None,
+    ) -> None:
+        self.max_size = max_size
+        self.mode = mode
+        self.buffering = buffering
+        self.encoding = encoding
+        self.newline = newline
+        self.suffix: AnyStr | None = suffix
+        self.prefix: AnyStr | None = prefix
+        self.dir: AnyStr | None = dir
+        self.errors = errors
+
+        self._async_file: AsyncFile | None = None
+
+    async def __aenter__(self) -> SpooledTemporaryFile:
+        fp = await to_thread.run_sync(
+            partial(
+                tempfile.SpooledTemporaryFile,
+                self.max_size,
+                self.mode,
+                self.buffering,
+                self.encoding,
+                self.newline,
+                self.suffix,
+                self.prefix,
+                self.dir,
+                errors=self.errors,
+            )
+        )
+        self._async_file = AsyncFile(fp)
+        return self
+
+    async def rollover(self) -> None:
+        if self._async_file is None:
+            raise RuntimeError("Internal file is not initialized.")
+        await to_thread.run_sync(cast(Callable[[], None], self._async_file.rollover))
```

Comment:
Add a blank line between the end of a control block and the next code line that follows it. Do this everywhere please.